### PR TITLE
[2.3] [Form] Modified iterator_to_array's 2nd parameter to false in ViolationMapper

### PR DESCRIPTION
**Pull request by issei-m**
| Q | A |
| --- | --- |
| Branch? | 2.3 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | n/a |
| License | MIT |
| Doc PR | n/a |

For https://github.com/symfony/symfony/pull/18747

### CODE REVIEW

File: src/Symfony/Component/Form/Extension/Validator/ViolationMapper/ViolationMapper.php [hunk 1]
```diff
@@ -188,7 +186,7 @@ private function matchChild(FormInterface $form, PropertyPathIteratorInterface $
             }
 
             /** @var FormInterface $child */
-            foreach ($children as $key => $child) {
+            foreach ($children as $i => $child) {
```
Comment (iltar) [inline src/Symfony/Component/Form/Extension/Validator/ViolationMapper/ViolationMapper.php]: Is there a specific reason for this change? If not, I like the original more. `$i` implies that it's being used for incrementation, but you're simply referring to the key here.

Comment (issei-m) [inline src/Symfony/Component/Form/Extension/Validator/ViolationMapper/ViolationMapper.php]: I have an intention that make the key of `$children` meaningless. (The name of `iterator_to_array`'s 2nd parameter is `$use_keys`)

Comment (iltar) [inline src/Symfony/Component/Form/Extension/Validator/ViolationMapper/ViolationMapper.php]: It's used to unset something below. if you mean to make it meaningless, you should remove it altogether

Comment (issei-m) [inline src/Symfony/Component/Form/Extension/Validator/ViolationMapper/ViolationMapper.php]: Moreover, `i` implies it's just counter indeed. But also implies it's an integer, IMO.

Comment (issei-m) [inline src/Symfony/Component/Form/Extension/Validator/ViolationMapper/ViolationMapper.php]: @alekitto How do you think about this?

Comment (alekitto) [inline src/Symfony/Component/Form/Extension/Validator/ViolationMapper/ViolationMapper.php]: I like the change: in my opinion is more clear that the variable contains an integer counter, not a meaningful key name.
Moreover it makes it consistent with the existing codebase, for example: `$i` is used in DI `Definition::removeMethodCall` method in a similar way.

Comment (issei-m) [inline src/Symfony/Component/Form/Extension/Validator/ViolationMapper/ViolationMapper.php]: @alekitto Thank you for advicing!

**Comment by fabpot** [comment]
Thank you @issei-m.

**Comment by fabpot** [comment]
@issei-m Can you submit a new PR on 2.7 to update the tests? When merging, I had to revert all tests changes due to conflicts hard to resolve easily :) Thanks.

**Comment by fabpot** [comment]
nevermind, I see the other PR now :)

**Comment by issei-m** [comment]
@fabpot Thanks for merging. For 2.7 is here: https://github.com/symfony/symfony/pull/18773.
btw will https://github.com/symfony/symfony/pull/18747 be unnecessary anymore?

**Comment by issei-m** [comment]
@fabpot Ah, OK! I will close https://github.com/symfony/symfony/pull/18773 ;)

**Pull request closed without merging**